spread.yaml

```diff
@@ -55,6 +55,29 @@ backends:
                 username: ubuntu
                 password: ubuntu
 
+    adhoc-ubuntu-core:
+        type: adhoc
+        environment:
+            SPREAD_ADHOC_UC_ADDRESS: "$(HOST: echo ${SPREAD_ADHOC_UC_ADDRESS:-localhost})"
+            SPREAD_ADHOC_UC_PORT: "$(HOST: echo ${SPREAD_ADHOC_UC_PORT:-8022})"
+        allocate: |
+            echo "Allocating $SPREAD_SYSTEM..."
+            ADDRESS $SPREAD_ADHOC_UC_ADDRESS:$SPREAD_ADHOC_UC_PORT
+        discard: |
+            echo "Discarding ad-hoc $SPREAD_SYSTEM"
+        restore: |
+            rm -rf $GOPATH/src
+        systems:
+            - ubuntu-core-16.04-64:
+                username: test
+                password: ubuntu
+            - ubuntu-core-16-device-amd64:
```

Comment:
There's a different system mainly because of the `tests/main/ack` test, according to Samuele it is expected to fail on the boards but not on a newly created image, he can explain better. With a different system we can exclude the task from being executed in the boards, please let me know if you have in mind a better way of doing it.

There are other tests, like `tests/main/prepare-image-uboot` or `tests/main/interfaces-network-control` that are excluded from the `ubuntu-core-16-64` system but work fine on the boards, maybe we could enable them in `ubuntu-core-16-64` too, i haven't dug into the reasons about why that was disabled.

Having this additional dummy `ubuntu-core-16.04-64` (note that it is different from `ubuntu-core-16-64`) system here prevents spread from crashing because of tasks without systems defined, when spread will be updated to handle those we can remove it, in the time being it has served to verify that all the rest of changes work as expected and to validate the upcoming images. I've added a `FIXME` notice so that we keep track of it.